util/conditions/experimental/merge_strategies.go

```diff
@@ -0,0 +1,408 @@
+	if isSummaryOperation {
+		messages := []string{}
+		for _, condition := range append(issueConditions, append(unknownConditions, infoConditions...)...) {
+			priority := getPriority(condition.Condition, negativeConditionTypes)
+			if priority == infoMergePriority {
+				// Drop info messages when we are surfacing issues on unknown.
+				if status != metav1.ConditionTrue {
+					continue
+				}
+				// Drop info conditions with empty messages.
+				if condition.Message == "" {
+					continue
+				}
+			}
+
+			m := fmt.Sprintf("%s (%s)", condition.Type, condition.Status)
+			if condition.Message != "" {
+				m += fmt.Sprintf(": %s", condition.Message)
+			}
+			messages = append(messages, m)
+		}
+
+		// Prepend the step counter if required.
+		if stepCounter {
+			totalSteps := len(conditionTypes)
+			stepsCompleted := len(infoConditions)
+
+			messages = append([]string{fmt.Sprintf("%d of %d completed", stepsCompleted, totalSteps)}, messages...)
+		}
+
+		message = strings.Join(messages, "; ")
+	}
+
+	// When performing the aggregate operation, we are merging one single condition from potentially many objects.
+	// Considering the high number of conditions involved, the messages from the conditions being merged must be filtered/summarized
+	// using rules designed to surface the most important issues.
+	//
+	// Accordingly, the resulting message is composed by only three messages from conditions classified as issues/unknown;
+	// instead three messages from conditions classified as info are included only if there are no issues/unknown.
+	//
+	// The number of objects reporting the same message determine the order used to pick the messages to be shown;
+	// For each message it is reported a list of max 3 objects reporting the message; if more objects are reporting the same
+	// message, the number of those objects is surfaced.
+	//
+	// e.g. (False): Message-1 from default/obj0, default/obj1, default/obj2 and 2 other Objects
+	//
+	// If there are other objects - objects not included in the list above - reporting issues/unknown (or info there no issues/unknown),
+	// the number of those objects is surfaced.
+	//
+	// e.g. ...; other 2 Objects with issues; other 1 Objects unknown
+	//
+	if isAggregateOperation {
+		// Gets the kind to be used when composing messages.
+		// NOTE: This assume all the objects in an aggregate operation are of the same kind.
+		kind := ""
+		if len(conditions) > 0 {
+			kind = flect.Pluralize(conditions[0].OwnerResource.Kind)
+		}
+
+		n := 3
+		messages := []string{}
+
+		// Get max n issue messages, decrement n, and track if there are other objects reporting issues not included in the messages.
+		if len(issueConditions) > 0 {
+			issueMessages, otherIssues := aggregateMessages(issueConditions, &n, kind, false)
+
+			messages = append(messages, issueMessages...)
+			if otherIssues > 0 {
+				messages = append(messages, fmt.Sprintf("other %d %s with issues", otherIssues, kind))
```

Comment:
Should we use something like "5 more <kind> with issues" (IIRC would match what we used for drain, and I think it sounds better)

Reply:
done!